### PR TITLE
feat(ListOrganizations): make it able to retrieve summary info of org…

### PR DESCRIPTION
**Pull request by takeshi-cloudnatix**
…anizations together

### CODE REVIEW

File: api/v1/user_manager_service.proto [hunk 1]
```diff
@@ -55,6 +55,13 @@ message Organization {
   string id = 1;
   string title = 2;
   int64 created_at = 3;
+
+  // project_count is the number of projects in the organization.
```
Comment (kkaneda) [inline api/v1/user_manager_service.proto:59]: Slightly prefer defining message like `Summary` add the two fields there.

File: server/internal/server/org.go [hunk 2]
```diff
@@ -109,17 +109,36 @@ func (s *S) ListOrganizations(ctx context.Context, req *v1.ListOrganizationsRequ
...
 	}
 
 	var orgProtos []*v1.Organization
-	for _, org := range filtered {
+	for _, org := range visibleOrgs {
 		orgProtos = append(orgProtos, org.ToProto())
 	}
+
+	if req.IncludeSummaries {
+		for _, org := range orgProtos {
```
Comment (kkaneda) [inline server/internal/server/org.go:125]: can't we add a quick test for this?

File: server/internal/server/org.go [hunk 3]
```diff
@@ -109,17 +109,36 @@ func (s *S) ListOrganizations(ctx context.Context, req *v1.ListOrganizationsRequ
...
+			ps, err := s.store.ListProjectsByTenantIDAndOrganizationID(userInfo.TenantID, org.Id)
```
Comment (kkaneda) [inline server/internal/server/org.go:126]: Actually not all the projects are visible to end users.. We would need to follow that logic we have in https://github.com/llmariner/user-manager/blob/main/server/internal/server/project.go#L143

**Review by kkaneda — COMMENTED** [review]
Thanks!

File: api/v1/user_manager_service.proto [hunk 4]
```diff
@@ -106,6 +117,7 @@ message CreateOrganizationRequest {
 }
 
 message ListOrganizationsRequest {
+  bool include_summaries = 1;
```
Comment (kkaneda) [inline api/v1/user_manager_service.proto:120]:
```suggestion
  bool include_summary = 1;
```

File: server/internal/server/org.go [hunk 5]
```diff
@@ -109,17 +109,44 @@ func (s *S) ListOrganizations(ctx context.Context, req *v1.ListOrganizationsRequ
...
+			}
+
+			var visiblePjCount int32
+			for _, p := range ps {
+				if s.validateProjectMember(p.ProjectID, p.OrganizationID, userInfo.UserID) == nil {
+					visiblePjCount++
+				}
+			}
+
+			us, err := s.store.ListOrganizationUsersByOrganizationID(org.Id)
```
Comment (kkaneda) [inline server/internal/server/org.go:138]: Can't we also call `validateOrganizationOwner` and show the user count only if the return value is `nil`? This follows https://github.com/llmariner/user-manager/blob/main/server/internal/server/org.go#L144

**Review by kkaneda — APPROVED** [review]
LG!

File: server/internal/server/org.go (hunk 5 above)
Comment (takeshi-cloudnatix) [reply server/internal/server/org.go]: done!

**Comment by takeshi-cloudnatix** [comment]
CC @saniya-cloudnatix

**Pull request merged**